### ISSUE TITLE
Bug: Type Name Collision

### ISSUE THREAD

**Issue by docteurklein** [supabase/pg_graphql#12]
# Bug report

I tried a query on an existing table (named "attribute") in schema "public" (and there is another table named the same in another schema).

I suspect it collides with pg_graphql's schema representation.

## Describe the bug

```
gql.field\n                    join gq
l.type type_\n                        on field.type_ = type_.name\n                where\n                    field.parent_type = 'Query
'\n                    and field.name = gql.name(ast_operation)\n                    and gql.is_visible(\"field\")\" returned more than
one row"
```

Once the conflicting table was removed, it started complaining about other unrelated tables to the "book" example:

```
select gql.resolve($$                                                                                              [1372/2149]
query {
  allBooks {
    edges {
      node {
        id
      }
    }
  }
}
$$);
                                       resolve
──────────────────────────────────────────────────────────────────────────────────────
 {"data": null, "errors": ["column \"name\" of relation \"product\" does not exist"]}
```
I had to remove all the "product" tables and other stuff I had in my schemas.

I'll try to debug it locally and see if I can find the cause.
Thanks for this formidable project :)

**Comment by olirice** [comment]
Thanks, that does sounds like a naming collision and its a use-case that needs to be supported

In the short-term you can get around that issue by creating a role that only has `SELECT` permissions to the tables you're interested in so there are no collisions when the queries are being built

[This test](https://github.com/supabase/pg_graphql/blob/4ec92c47640498b8744ffc03c57ea2082f8c5543/test/expected/permissions_connection_column.out#L22) may be a useful reference

**Comment by olirice** [comment]
RFC #18 
and WIP implementation #19

**Comment by olirice** [comment]
There is now a method for overriding type and field names described in the  [configuration docs](https://supabase.github.io/pg_graphql/configuration/)

I experimented with enforcing uniqueness of type names but found that it was fiddly while evolving schemas. The schema reloads using an event trigger after ddl statements so a migration like this would crash

```sql
create schema app;

create table public.account(
    id serial primary key
);

create table app.account( -- Error: Unique Constraint Violated
    id serial primary key
);
```

Additionally, the name collision issue isn't global. If a user's role does not have select privileges for `app.account`, it would be omitted from the graphql schema and there would be no problem.

For those reasons, I've decided not to prohibit name collisions in the schema in favor of directing users to the override system, or role based permissions to filter duplicates out
